code/game/objects/items/weapons/clown_items.dm

```diff
@@ -121,12 +121,13 @@
 
 /obj/item/weapon/bikehorn/attack_self(mob/user)
 	if(!spam_flag)
-		spam_flag = 1
+		spam_flag = TRUE
 		playsound(src.loc, honksound, 50, 1)
 		src.add_fingerprint(user)
-		spawn(cooldowntime)
-			spam_flag = 0
-	return
+		addtimer(CALLBACK(src, .proc/clear_cooldown), cooldowntime)
+
+/obj/item/weapon/bikehorn/proc/clear_cooldown()
+	spam_flag = FALSE

```

Comment:
@RemieRichards what would be less intensive, using a timer for the spam flag, or using a world.time timer? Keep in mind the use case of spamming the horn a lot, and I assume that checking a bool is faster than comparing two numbers?